apps/etl-func/src/adapters/functions/message-ingestion-error-queue.ts

```diff
@@ -0,0 +1,42 @@
+import { IngestMessageUseCase } from "@/domain/use-cases/ingest-message.js";
+import { StorageQueueHandler } from "@azure/functions";
+import { pino } from "pino";
+
+import {
+  MessageMetadata,
+  messageMetadataSchema,
+} from "../../domain/message.js";
+
+const logger = pino({
+  level: process.env.NODE_ENV === "production" ? "error" : "debug",
+});
+
+const processMessageMetadata = (
+  input: unknown,
+): MessageMetadata | undefined => {
+  const result = messageMetadataSchema.safeParse(input);
+  //we are considering, and returning, only processed messages. A processed message has isPending === false
+  if (result.success && !result.data.isPending) return result.data;
+  else return undefined;
+};
+
+const messagesIngestionErrorQueueHandler =
+  (ingestUseCase: IngestMessageUseCase): StorageQueueHandler =>
+  async (queueItem: unknown) => {
+    //converting single object to an array to use the same ingestion logic
+    //it will always be an 1 item array
+    const queueItemArray = [...[], queueItem];
+    const parsedMessagesMetadata = queueItemArray.map(processMessageMetadata);
+    const messagesMetadata: MessageMetadata[] = parsedMessagesMetadata.filter(
+      (item): item is MessageMetadata => item !== undefined,
+    );
+
+    try {
+      await ingestUseCase.execute(messagesMetadata);
```

Comment:
I understand that you want to create an array in order to reuse the ingestUseCase. On the other hand, if we want to use this approach, I think it would be cleaner to parse the single item first and then create the array.
This way, you don't have to call any map. You can also remove the filter if you perform the check beforehand. I would write something similar.

```suggestion
const messagesIngestionErrorQueueHandler =
  (ingestUseCase: IngestMessageUseCase): StorageQueueHandler =>
  async (queueItem: unknown) => {
    const parsedMetadata = processMessageMetadata(queueItem);
     if (parsedMetadata === undefined) return;

    try {
      await ingestUseCase.execute([parsedMetadata]);
```

Reply:
2a0ced619b90256f3803819eed888f5f8731b29e